python/lsst/ts/tcpip/base_client_or_server.py

```diff
@@ -332,6 +333,32 @@ async def read_into(self, struct: ctypes.Structure) -> None:
             await self._close_client()
             raise
 
+    async def read_str(self) -> str:
+        """Read data ending in `TERMINATOR`.
+
+        Before returning the data, it is decoded into an UTF-8 string.
```

Comment:
I gently suggest stripping terminator. I realize readline type functions typically don't, but they also require the user to *specify* the terminator, so the user knows what to strip. In this case the terminator is an internal detail that the user should not need to know, and thus cannot as easily strip (the user does not even know how many chars are in the terminator).

Reply:
Good suggestion. Stripping the TERMINATOR only isn't straightforward, since it is a binary string so it has to be decoded first. I don't think it would be a good idea to simply use the `strip()` method without indicating what to strip since that could strip more than just the TERMINATOR. So I ended up doing

```
return data.decode(encoding="utf-8", errors="strict").strip(
    TERMINATOR.decode(encoding="utf-8", errors="strict")
)
```

Reply:
I think you can also just chop off `len(TERMINATOR)` chars at the end of the bytes array, e.g.:
```
term_len = len(TERMINATOR)
return bytes_data[0:-term_len].decode(...)
```